web/modules/custom/joinup_community_content/templates/link-icon.html.twig

```diff
@@ -0,0 +1,14 @@
+{#
+/**
+ * @file
+ * Default theme implementation of link that are accompanied with an icon.
+ *
+ * Available variables:
+ * - icon: The name of the icon.
+ * - title: The link title.
+ * - url: The link URL.
+ *
+ * @ingroup themeable
+ */
+#}
+<a href="{{ url }}"{{ attributes }}><span class="action-link__icon icon icon--{{ icon }}"></span><span class="action-link__link">{{ title }}</span></a>
```

Comment:
Normally a `<a ...>` tag can is used inline but this template will add a new line at the end. Let's use `{% spaceless %}`.